open_lm/attention.py

```diff
@@ -125,8 +125,12 @@ def get_attn_func(
     attn_seq_scalar=None,
     alpha=None,
 ):
-    if attn_name == "xformers_attn":
+    if attn_name == "auto":
+        return xformers_attn if torch.cuda.is_available() else torch_attn
+    elif attn_name == "xformers_attn":
         return xformers_attn
+    elif attn_name == "xformers_attn_with_variable_length_support":
+        return lambda *args, **kwargs: xformers_attn(*args, **kwargs).contiguous()
```

Comment:
can we add a comment here on what this code path does / when to use it e.g.,

native xformers_attention assumes that inputs are contiguous. the call to contiguous here ensures that intermediate activations remain contiguous even for variable length inputs that are not powers of 2 (e.g., sequence lengths of 51 etc.)